### ISSUE TITLE
Fast way to append to huge csv file

### ISSUE THREAD

**Issue by Fuco1** [johnkerl/miller#278]
I have 10k jobs each of which produces a 20 megabyte file.  This is 200gb of data in total.  I need to concat all these files into one huge output file.  As they are produced in parallel maybe 500 at a time, I would like to immediately start processing the files which are ready.

Is there a good way to do a "cat" which would only append one csv to another, without having to read the entire file first?

**Comment by johnkerl** [comment]
Hmm ... well `cat` is super-optimized. Have you timed how costly `cat` is being for you here?

Also, just curious, why do those need to be one huge output file?

**Comment by Fuco1** [comment]
I'm currently using this 

```
mlr --no-mmap --tsv --nr-progress-mod 50000 \
            put '$date = strftime(strptime($date, "%Y-%m-%dT%H:%M:%SZ"), "%Y-%m-%d")' \
            output-*.tsv 
```

basically only to convert the date, but I could maybe change the generators to do it properly right away.

The final file is then downloaded through HTTP by another system.  I do not own this system and can't change it.

The above takes several hours to complete, so I want a solution where I can basically generate the big file at the same time as the batch jobs are finishing (we have 500 workers for 10-50k jobs).  Since this whole process takes about 4 hours that would be enough time to "in parallel" also concat the files with miller into the output.

I'm afraid that running `mlr` with `cat huge-file new-file` will do too much seeking and get really slow especially when the huge-file grows over a certain limit.  

This also happens in the cloud with not-that-stellar IO performance.

I guess there's not much we can do with regular file system on linux so this might be outside of the scope of this project.

**Comment by johnkerl** [comment]
Yes, having the OS put multiple files together is done by `/bin/cat` and I would expect nothing faster could be done.

If you want very high throughput then I would recommend against `mlr cat`. Rather, `sed 1d foo.csv` will omit the header line (very performantly) and you can use a combination of `sed` and `cat` to concatenate your CSV files quicker than `mlr cat` can.

**Comment by Fuco1** [comment]
Thanks for the feedback John.  Yes, I suspected so and that's the solution we opted for (I used `tail` instead of `sed`... I really should learn `sed` better ;).

I also consulted our data people and we did some changes in the output to make it much smaller e.g. for columns which were enums we print 0,1,2,3... instead of the long names, which reduced the size of the output by good 70%.

Cheers and thanks again for your time!
